### DISCUSSION TITLE
Can't use with Vogen

### DISCUSSION THREAD

**Discussion by aradalvand** [AArnott/Nerdbank.MessagePack#98] [General]
Vogen is a library that [source-generates value objects](https://github.com/SteveDunn/Vogen?tab=readme-ov-file#overview), since PolyType (which Nerdbank.MessagePack is based on) is also a source generator, and .NET doesn't support chaining source generators, PolyType won't be able to see (and therefore generate metadata) for the stuff that Vogen generates, meaning that Nerdbank.MessagePack cannot be used with Vogen or any source generator library that augments types, for that matter.

**Comment by eiriktsarpalis** [comment]
Question: how does Vogen compare to using record types? Seems related in terms of scope.

**Reply by aradalvand** [reply]
https://github.com/SteveDunn/Vogen?tab=readme-ov-file#why-cant-i-just-use-public-record-struct-customeridint-value

**Reply by eiriktsarpalis** [reply]
Fair. I don't think it's possible to address this at a library level. From a source generator architecture standpoint, it is not possible to layer generator outputs.

**Comment by AArnott** [comment]
Since you're coming from a MessagePack-CSharp perspective, I suspect you were already using a Reflection-based approach to serialization (i.e. not using MessagePack v3's source generator or v2's mpc tool). 
In that case, Nerdbank.MessagePack _does_ have an equivalent mode that you can use. Check out [this doc](https://aarnott.github.io/Nerdbank.MessagePack/docs/type-shapes.html#fallback-configuration) for more information on using the reflection provider instead of the source generator.
